### ISSUE TITLE
Failed opening required 'recipe/laravel-deployer.php'

### ISSUE THREAD

**Issue by Strift** [lorisleiva/laravel-deployer#23]
Hello,

The error:
```bash
PHP Warning:  require(recipe/laravel-deployer.php): failed to open stream: No such file or directory in /home/strift/Code/LaKazCreole/vendor/lorisleiva/laravel-deployer/.build/deploy.php on line 5
PHP Fatal error:  require(): Failed opening required 'recipe/laravel-deployer.php' (include_path='phar:///home/strift/Code/LaKazCreole/vendor/deployer/dist/dep/bin/../:.:/usr/share/php') in /home/strift/Code/LaKazCreole/vendor/lorisleiva/laravel-deployer/.build/deploy.php on line 5
```

Steps to reproduce:
```bash
composer require lorisleiva/laravel-deployer
php artisan deploy:init
php artisan deploy 
```

You have the same error with any other commands too.

Hope you can fix this quick so I can start trying the tool! It looks both promising and welcomed in the Laravel ecosystem.

Cheers,

**Comment by lorisleiva** [comment]
Hi there, thank you for reporting this. It looks like the include path does not include the Laravel Deployer folder. Could you please tell me which version of `deployer/deployer` you are using (in your composer.lock)?

**Comment by Strift** [comment]
Hi,

Deployer version:

```json
"name": "deployer/deployer",
            "version": "v6.2.0",
```

Default config/deploy.php file:
```php
<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Default deployment strategy
    |--------------------------------------------------------------------------
    |
    | This option defines which deployment strategy to use by default on all
    | of your hosts. Laravel Deployer provides some strategies out-of-box
    | for you to choose from explained in detail in the documentation.
    |
    | Supported: 'basic', 'firstdeploy', 'local', 'pull'.
    |
    */

    'default' => 'basic',

    /*
    |--------------------------------------------------------------------------
    | Custom deployment strategies
    |--------------------------------------------------------------------------
    |
    | Here, you can easily set up new custom strategies as a list of tasks.
    | Any key of this array are supported in the `default` option above.
    | Any key matching Laravel Deployer's strategies overrides them.
    |
    */

    'strategies' => [
        //
    ],

    /*
    |--------------------------------------------------------------------------
    | Hooks
    |--------------------------------------------------------------------------
    |
    | Hooks let you customize your deployments conveniently by pushing tasks
    | into strategic places of your deployment flow. Each of the official
    | strategies invoke hooks in different ways to implement their logic.
    |
    */

    'hooks' => [
        // Right before we start deploying.
        'start' => [
            //
        ],

        // Code and composer vendors are ready but nothing is built.
        'build' => [
            'npm:install',
            'npm:production',
        ],

        // Deployment is done but not live yet (before symlink)
        'ready' => [
            'artisan:storage:link',
            'artisan:view:clear',
            'artisan:cache:clear',
            'artisan:config:cache',
            'artisan:optimize',
            'artisan:migrate',
        ],

        // Deployment is done and live
        'done' => [
            'fpm:reload',
        ],

        // Deployment succeeded.
        'success' => [
            //
        ],

        // Deployment failed.
        'fail' => [
            //
        ],
    ],

    /*
    |--------------------------------------------------------------------------
    | Deployment options
    |--------------------------------------------------------------------------
    |
    | Options follow a simple key/value structure and are used within tasks
    | to make them more configurable and reusable. You can use options to
    | configure existing tasks or to use whithin your own custom tasks.
    |
    */

    'options' => [
        'application' => env('APP_NAME', 'Laravel'),
        'repository' => 'git@github.com:mygithub/myrepo.git',
        'php_fpm_service' => 'php7.2-fpm',
    ],

    /*
    |--------------------------------------------------------------------------
    | Hosts
    |--------------------------------------------------------------------------
    |
    | Here, you can define any domain or subdomain you want to deploy to.
    | You can provide them with roles and stages to filter them during
    | deployment. Read more about how to configure them in the docs.
    |
    */

    'hosts' => [
        'mydomain.com' => [
            'deploy_path' => '/home/deployer/lakazcreole',
            'user' => 'deployer',
            'stage' => 'production',
        ],
    ],

    /*
    |--------------------------------------------------------------------------
    | Localhost
    |--------------------------------------------------------------------------
    |
    | This localhost option give you the ability to deploy directly on your
    | local machine, without needing any SSH connection. You can use the
    | same configurations used by hosts to configure your localhost.
    |
    */

    'localhost' => [
        //
    ],

    /*
    |--------------------------------------------------------------------------
    | Include additional Deployer recipes
    |--------------------------------------------------------------------------
    |
    | Here, you can add any third party recipes to provide additional tasks,
    | options and strategies. Therefore, it also allows you to create and
    | include your own recipes to define more complex deployment flows.
    |
    */

    'include' => [
        //
    ],

    /*
    |--------------------------------------------------------------------------
    | Use a custom Deployer file
    |--------------------------------------------------------------------------
    |
    | If you know what you are doing and want to take complete control over
    | Deployer's file, you can provide its path here. Note that, without
    | this configuration file, the root's deployer file will be used.
    |
    */

    'custom_deployer_file' => false,

];
```

I tried re-running it after `composer dump-autoload`. It still isn't working. :/

**Comment by lorisleiva** [comment]
This is very odd. 😕

I don't understand why the vendor folder of Laravel Deployer is not part of the `include_path`. Is your application manipulating namespaces in an unusual way?

**Comment by Strift** [comment]
Not that I know of. It's a pretty standard Laravel app.

I tried with a default Laravel app though, and it works fine. So it sure has something to do with how my app works, but I could not say what. It is a pretty basic Laravel app, with no fancy dependency (debugbar, and a few more packages necessary for some Laravel features).

In case it could help:
```json
   "require": {
        "php": ">=7.0.0",
        "doctrine/dbal": "^2.6",
        "fideloper/proxy": "~4.0",
        "guzzlehttp/guzzle": "^6.3",
        "laravel/framework": "5.6.*",
        "laravel/tinker": "~1.0",
        "lorisleiva/laravel-deployer": "^0.2.3",
        "rap2hpoutre/laravel-log-viewer": "^0.14.3",
        "spatie/laravel-tail": "^3.0"
    },
    "require-dev": {
        "barryvdh/laravel-debugbar": "^3.1",
        "caouecs/laravel-lang": "~3.0",
        "deployer/dist": "^6.0",
        "deployer/recipes": "^6.0",
        "filp/whoops": "~2.0",
        "fzaninotto/faker": "~1.4",
        "mockery/mockery": "~1.0",
        "phpunit/phpunit": "~7.0",
        "symfony/thanks": "^1.0"
    },
```

Maybe is it because I had a deploy installation first?

**Comment by lorisleiva** [comment]
Yes the only thing that I can think of is that `deployer/dist`and potentially `deployer/recipes` are conflicting with the dependencies used by Laravel Deployer. Could you please try to deploy without those two dependencies?

**Comment by Strift** [comment]
I removed them both and it worked.

I should have removed them anyway so it doesn't bother me, but it still is strange that the error occur.

Thanks for the help!

**Comment by lorisleiva** [comment]
Okay cool! I think you could still use `deployer/recipes` as it just adds a new path on the include path like Laravel Deployer does. However `deployer/dist` basically replaces the entire binary so that must definitely cause some problems. I'm glad I could help 🙂
